From 1357576ab6f40a6bc9674d64355843cc9408d9cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?P=C3=A1draig=20Brady?= <pbrady@redhat.com>
Date: Wed, 17 Jul 2013 16:27:31 +0100
Subject: [PATCH] avoid code path causing qpid exchange leaks

Always assume amqp_rpc_single_reply_queue is True,
so that the problematic code path is ignored.
The issue is discussed at https://pad.lv/1178375
---
 quantum/openstack/common/rpc/amqp.py |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/quantum/openstack/common/rpc/amqp.py b/quantum/openstack/common/rpc/amqp.py
index dbfc790..c7ce7c7 100644
--- a/quantum/openstack/common/rpc/amqp.py
+++ b/quantum/openstack/common/rpc/amqp.py
@@ -590,7 +590,8 @@ def multicall(conf, context, topic, msg, timeout, connection_pool):
 
     # TODO(pekowski): Remove this flag and the code under the if clause
     #                 in Havana.
-    if not conf.amqp_rpc_single_reply_queue:
+    # (p-draigbrady): This clause is disabled to avoid qpid exchange leaks
+    if False and not conf.amqp_rpc_single_reply_queue:
         conn = ConnectionContext(conf, connection_pool)
         wait_msg = MulticallWaiter(conf, conn, timeout)
         conn.declare_direct_consumer(msg_id, wait_msg)
